docs/rfcs/20210223_external_connectivity_in_k8s.md

```diff
@@ -0,0 +1,230 @@
+- Feature Name: Rafał Korepta
+- Status: in-progress
+- Start Date: 2021-02-23
+- Authors: RafalKorepta
+- Issue: (one or more # from the issue tracker)
+
+# Executive Summary
+
+Users would like to managed Redpanda clusters in kubernetes environment,
+but the clients could live outside of the kubernetes network boundary.
+To give users out of the box experience Redpanda kubernetes operator
+should pave the way for the networking stack to expose each Redpanda
+node in k8s environment.
+
+## What is being proposed
+
+The most performant solution is to expose each process to the outside,
+by specifying
+[host port in the container ports specification](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#containerport-v1-core)
+for each Redpanda node. Each instance will
+need to be registered in DNS provider of one's choice.
+
+The `external-dns` project is chosen as a integration point to ease
+the integration with the DNS provider.
+
+## Why (short reason)
+
+The Redpanda is focused on performance and its low tail latency. To
+not add more network hops this proposal tries to merit dynamic
+kubernetes environment with the direct access to Redpanda process.
+
+## How (short plan)
+
+The [PR #658](https://github.com/vectorizedio/redpanda/pull/658)
+introduces Service type NodePort and DNSEndpoint resource. The
+Service resources focus on exposing unique ports in each kubernetes
+node. Operator responsibility is to create statefulset that managed
+Redpanda instances in a form of kubernetes Pod. After statefulset
+creates pods, the operator will map assigned node public IP to
+the unique broker DNS. To achieve that the DNSEndpoint custom
```

Comment:
node public IP to the unique broker DNS -> DNS name perhaps? If I understand this correctly

Reply:
Thanks, fixed.